src/components/walletconnect/WalletConnectModals.js

```diff
@@ -0,0 +1,341 @@
+  const theme = useTheme()
+  const { colors, sizes } = theme
+  const { borderRadius } = sizes
+  const { primary } = colors
+
+  const show = useCallback(({ session, payload, message, walletAddress, onReject, onApprove, modalType, explorer }) => {
+    log.debug('showing dialog', { session, payload, message, walletAddress, onReject, onApprove, modalType })
+    if (modalType === 'error') {
+      return showErrorDialog(t`Unsupported request ${payload.method}`)
+    }
+
+    try {
+      showDialog({
+        showCloseButtons: false,
+        isMinHeight: false,
+        showButtons: true,
+        content: (
+          <ApproveModal
+            payload={payload}
+            message={message}
+            session={session}
+            walletAddress={walletAddress}
+            modalType={modalType}
+            explorer={explorer}
+            onScan={data => {
+              if (!data) {
+                return
+              }
+              const ok = onApprove(data)
+              if (!ok) {
+                showErrorDialog(t`Invalid QR Value: ${data}`)
+              }
+              hideDialog()
+            }}
+          />
+        ),
+        buttonsContainerStyle: {
+          width: '95%',
+          flexDirection: 'row',
+          justifyContent: 'space-between',
+          marginTop: sizes.defaultDouble,
+        },
+        buttons: [
+          {
+            text: 'Reject',
+            onPress: async dismiss => {
+              // do something
+              try {
+                await onReject()
+                dismiss()
+              } catch (e) {
+                log.error('failed rejecting', e.message, e, { dialogShown: true, payload, modalType })
+                showErrorDialog(t`Could not reject request.`)
+              }
+            },
+            color: 'red',
+            style: {
+              width: '48%',
+              color: 'blue',
+              marginRight: 10,
+              borderRadius,
+            },
+          },
+          {
+            text: 'Approve',
+            onPress: async dismiss => {
+              // do something
+              try {
+                await onApprove()
+                dismiss()
+              } catch (e) {
+                log.error('failed approving', e.message, e, { dialogShown: true, payload, modalType })
+                showErrorDialog(t`Could not approve request.`)
+              }
+            },
+            color: 'white',
+            style: {
+              borderRadius,
+              width: '48%',
+              backgroundColor: primary,
+              display: modalType !== 'scan' ? 'block' : 'none',
+            },
+          },
+        ],
+      })
+    } catch (e) {
+      log.error('failed showing dialog', e.message, e, { dialogShown: true, payload, modalType })
+
+      showErrorDialog(t`Unable to process request: ${e.message}`)
+    }
+  }, [])
+
+  return { show, isDialogShown }
```

Comment:
too much stuff in a single file
i suggest to rework structure as the following

```
walletconnect
    components
        ContractCall.js
        <other modals / components could be placed i a single file>
    screens
        WalletConnectScan.js
    hooks
        useSessionApproveModal.js
    theme
        withStyles.js <defines getStylesFromProps, exports withStyles(getStylesFromProps) HoC>
        

```

Reply:
over kill

Reply:
@sirpy nope, it's a general approach in the app design, the single responsibility principle from SOLID
in react it means 

1. one file - one exported component (small/helpers sub-components are also allowed if need)
2. one file - one hook (ideally). or a set of a few small hook with similar designation 

additionally
a) common styles could be moved in some file from where they will be imported
b) it's better to move top level components to the separate folder as 'screens'

please see how FV is implemented and structured
this is one of 2 my fundamental requests here

Reply:
if you have most important tasks todo and cannot proceed with this, you could just ask me. i will refactor

Reply:
you decided single responsibility means single file.  I actually see it the other way around, that things that have a single purpose should be in one file. i hate the per hook file pattern.
again it makes sense if this is a hook thats going to be used through out the app, if it is something that's specific to walletconnect there's no reason for the per hook per component per file. It is not a god given law.
from my point of view FV is unreadable, to this day i'm not able to find anything with all the jumping I need to do between components, files, classes and so on.

Reply:
structured code have own benefits. it's detailed, hierarchical, easy to split and understand logic, easy to find a bug
i understand this may be subjective, but for me is very hard to understand large files with a lot of stuff there

let's do the following compromise 
- approve + hook first file
- other stuff second file

Reply:
@sirpy ^^^